filters/auth/grant_test.go

```diff
@@ -0,0 +1,466 @@
+package auth_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters/auth"
+	"github.com/zalando/skipper/filters/builtin"
+	"github.com/zalando/skipper/proxy/proxytest"
+	"github.com/zalando/skipper/routing"
+	"github.com/zalando/skipper/secrets"
+	"golang.org/x/oauth2"
+)
+
+const (
+	testToken                = "foobarbaz"
+	testRefreshToken         = "refreshfoobarbaz"
+	testAccessCode           = "quxquuxquz"
+	testSecretFile           = "testdata/authsecret"
+	testAccessTokenExpiresIn = int(time.Hour / time.Second)
+	testCookieName           = "testcookie"
+)
+
+func newGrantTestTokeninfo(validToken string, tokenInfoJSON string) *httptest.Server {
+	const prefix = "Bearer "
```

Comment:
we can reuse `authHeaderPrefix`

Reply:
I left this as is. The reason is that `authHeaderPrefix` lives in a different package (`auth`) than the test (`auth_test`). Or should these be in the same package?

Also, it might be better to validate against the expected string `Bearer ` here in order to break the test if someone accidentally modifies the constant to an invalid value.

Reply:
You are right, I missed the package _test.